cumulusci/tasks/tests/test_connectedapp.py

```diff
@@ -0,0 +1,269 @@
+""" Tests for the connectedapp tasks """
+
+import logging
+import mock
+import os
+import pytest
+import re
+import unittest
+
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:
+    JSONDecodeError = ValueError
+from mock import MagicMock
+from mock import patch
+
+import cumulusci.core.tasks
+from cumulusci.core.config import (
+    BaseGlobalConfig,
+    BaseProjectConfig,
+    TaskConfig,
+    OrgConfig,
+    ServiceConfig,
+)
+from cumulusci.core.exceptions import ServiceNotConfigured, TaskOptionsError
+from cumulusci.core.keychain import BaseProjectKeychain
+from cumulusci.core.tests.utils import MockLoggingHandler
+from cumulusci.tasks.salesforce.tests.util import create_task
+from cumulusci.tasks.connectedapp import CreateConnectedApp
+from cumulusci.utils import temporary_dir
+
+
+class TestCreateConnectedApp(unittest.TestCase):
+    """ Tests for the CreateConnectedApp task """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestCreateConnectedApp, cls).setUpClass()
+        logger = logging.getLogger(cumulusci.core.tasks.__name__)
+        logger.setLevel(logging.DEBUG)
+        cls._task_log_handler = MockLoggingHandler(logging.DEBUG)
+        logger.addHandler(cls._task_log_handler)
```

Comment:
This logging stuff should go in a mixin rather than repeating it in multiple testcase classes.